wasm2c/wasm-rt.h

```diff
@@ -60,52 +60,165 @@ extern "C" {
 #endif
 
 /**
- * Enable memory checking via a signal handler via the following definition:
+ * Backward compatibility: Convert the previously exposed
+ * WASM_RT_MEMCHECK_SIGNAL_HANDLER macro to the ALLOCATION and CHECK macros that
+ * are now used.
+ */
+#if defined(WASM_RT_MEMCHECK_SIGNAL_HANDLER)
+
+#if WASM_RT_MEMCHECK_SIGNAL_HANDLER
+#define WASM_RT_MEMORY_ALLOCATION_MMAP 1
+#define WASM_RT_MEMORY_CHECK_GUARD 1
+#else
+#define WASM_RT_MEMORY_ALLOCATION_MALLOC 1
+#define WASM_RT_MEMORY_CHECK_BOUNDS 1
+#endif
+
+#warning \
+    "WASM_RT_MEMCHECK_SIGNAL_HANDLER has been deprecated in favor of WASM_RT_MEMORY_ALLOCATION_ and WASM_RT_MEMORY_CHECK_ macros"
+#endif
+
+/**
+ * Set the allocation strategy for Wasm memories.
+ *
+ * MMAP: the memory allocation and growth is performed with mmap and mprotect
+ * (or their Windows equivalents) respectively.
  *
- * #define WASM_RT_MEMCHECK_SIGNAL_HANDLER 1
+ * MALLOC: the memory allocation and growth is performed with malloc and realloc
+ * respectively.
  *
- * This is usually 10%-25% faster, but requires OS-specific support.
+ * This defaults to MMAP on 64-bit platforms (so we can use the GUARD based
+ * range checks below) and MALLOC on 32-bit platforms to minimize memory
+ * consumption.
  */
+#ifndef WASM_RT_MEMORY_ALLOCATION_MMAP
+#define WASM_RT_MEMORY_ALLOCATION_MMAP 0
+#endif
 
-#ifndef WASM_RT_SKIP_SIGNAL_RECOVERY
-#define WASM_RT_SKIP_SIGNAL_RECOVERY 0
+#ifndef WASM_RT_MEMORY_ALLOCATION_MALLOC
+#define WASM_RT_MEMORY_ALLOCATION_MALLOC 0
```

Comment:
As far as I can tell this is never actually used.   How about we just have a single macro (`WASM_RT_MEMORY_ALLOCATION_MMAP`) with malloc being used when its not set.   We could even use a shorter name like `WASM_RT_USE_MMAP`?

Reply:
Sure, I can make this change!